### PR TITLE
Add Manhattan and Minkowski distances to KNN with doctests

### LINKED ISSUE #13546: Enhancement : Manhattan Distance and Minkowski Distance for K Nearest Neighbours
### Description

The current implementation of the K-Nearest Neighbors (KNN) algorithm in `machine_learning/k_nearest_neighbors/` uses the Euclidean Distance by default.

To enhance the algorithm's functionality and completeness, this issue proposes adding two additional distance metrics that are commonly used in the KNN algorithm:

1.  **Manhattan Distance (L1 norm)**
2.  **Minkowski Distance (Lp norm)**, which is a generalized form that includes both Euclidean (p=2) and Manhattan (p=1) as special cases.

### Proposed changes

1.  Modify the KNN implementation (`k_nearest_neighbors.py` or similar file) to include a new parameter, perhaps `distance_metric` (e.g., set to 'euclidean' by default), that accepts 'manhattan' and 'minkowski' as values.
2.  Implement the calculation for the **Manhattan Distance**.
3.  Implement the calculation for the **Minkowski Distance**, using an additional parameter `p` for the power (e.g., defaulting to `p=2` if `distance_metric` is 'minkowski').
4.  Add comprehensive unit tests to ensure that the algorithm performs correctly when using these new distance metrics.

### PR DESCRIPTION
**Pull request by kdt523**
This PR extends the K-Nearest Neighbours (KNN) implementation to support additional distance metrics while keeping Euclidean as the default.
What’s included:
Adds distance_metric parameter to KNN: 'euclidean' (default), 'manhattan', 'minkowski'
Adds p parameter for Minkowski (Lp) distance (validated p >= 1)
Implements _manhattan_distance and _minkowski_distance helpers
Updates classify() to use the selected metric
Stores training data as a list to avoid zip exhaustion on multiple calls
Moves the scikit-learn demo under main so importing the module has no extra dependency
Doctests: distances, classification across metrics, and error handling
Backward compatibility:
Existing code remains unchanged by default
(distance_metric='euclidean').
No documentation changes.
How to verify locally:
From the Python folder:
Doctests:
python -m doctest -v [k_nearest_neighbours.py]
Fixes #13546

### CODE REVIEW

**Comment by kdt523** [comment]
Hi @TheAlgorithms/maintainers 👋
This PR Add Manhattan and Minkowski distances to KNN with doctests #13555 
Could someone with write access please review and approve it when you get a chance?  
Thanks a lot for your time and help!

**Comment by kdt523** [comment]
Hi @TheAlgorithms/maintainers 
This PR   #13555 Add Manhattan and Minkowski distances to KNN with doctests 
Could someone with write access please review and approve it when you get a chance?  
Thanks a lot for your time and help!

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/blob/master/maths/manhattan_distance.py
https://github.com/TheAlgorithms/Python/blob/master/maths/minkowski_distance.py
